app/src/org/commcare/utils/FileUtil.java

```diff
@@ -544,6 +544,14 @@ public static String getFileName(Context context, Uri uri) throws FileExtensionN
                 cursor.moveToFirst();
                 fileName = cursor.getString(nameIndex);
                 cursor.close();
+                // If file name doesn't have an extension, we should try UriToFilePath

```

Comment:
I don't like the way I've written this. 
I think we can change this and try to get fileName from `UriToFilePath` first, and then use `ContentResolver` and then `MimeTypeMap`.